pkg/operator/ceph/controller/spec.go

```diff
@@ -75,22 +75,24 @@ var (
 	cronLogRotate = `
 CEPH_CLIENT_ID=%s
 PERIODICITY=%s
-LOG_ROTATE_CEPH_FILE=/etc/logrotate.d/ceph
-
-if [ -z "$PERIODICITY" ]; then
-	PERIODICITY=24h
-fi
+LOG_ROTATE_CEPH_FOLDER=/etc/logrotate.d/ceph
+DAEMON_LOG_FILE=/var/log/ceph/$CEPH_CLIENT_ID.log
+LOG_MAX_SIZE=%s
 
 # edit the logrotate file to only rotate a specific daemon log
 # otherwise we will logrotate log files without reloading certain daemons
 # this might happen when multiple daemons run on the same machine
-sed -i "s|*.log|$CEPH_CLIENT_ID.log|" "$LOG_ROTATE_CEPH_FILE"
+sed -i "s|*.log|$CEPH_CLIENT_ID.log|" "$LOG_ROTATE_CEPH_FOLDER"
+
+# replace default daily with given user input
+sed --in-place "s/daily/$PERIODICITY/g" "$LOG_ROTATE_CEPH_FOLDER"
+
+# adding maxsize $LOG_MAX_SIZE at the 4th line of the logrotate config file with 4 spaces to maintain indentation
+sed --in-place "4i \ \ \ \ maxsize $LOG_MAX_SIZE" "$LOG_ROTATE_CEPH_FOLDER"
```

Comment:
Does 4th line really matter? Can't it simply be after the first line instead?

Reply:
Also I thought this comment https://github.com/rook/rook/pull/10528#issuecomment-1189859094 said we won't change nor set `maxsize` if the user is not setting it.

Reply:
> Does 4th line really matter? Can't it simply be after the first line instead?

I don't think so, but looking to the config file example and indentation I followed that